src/storage/cloud-extensions.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  Bucket,
+  BucketOptions,
+  File,
+  FileOptions,
+  Storage as StorageClient,
+  StorageOptions,
+} from "@google-cloud/storage";
+import { FirebaseError } from "../utils/error";
+interface FirebaseMetadata {
+  name: string;
+  bucket: string;
+  generation: string;
+  metageneration: string;
+  contentType: string;
+  timeCreated: string;
+  updated: string;
+  storageClass: string;
+  size: string;
+  md5Hash: string;
+  contentEncoding: string;
+  contentDisposition: string;
+  crc32c: string;
+  etag: string;
+  downloadTokens?: string;
+}
+/**
+ * An extension of the existing Google Cloud File class that allows users to utilize firebase-specific features.
+ */
+export class FirebaseStorageFile extends File {
+  constructor(
+    bucket: FirebaseStorageBucket,
+    name: string,
+    private endpoint: string,
```

Comment:
Firestore just uses the google cloud package and does something like this:
``
 const settings: Settings = {
  host: "localhost:8080",
  ssl: false
};

// Initialize Firestore
const firestore = new Firestore(settings);
```
For RTDB, we ask them to override the databaseUrl:
```
admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
    databaseURL: "https://movie-picker-729bb-default-rtdb.firebaseio.com",
  });
 ```

So I don't think there's a best practices way to get this done. My recommendation is to do something like:
```
getStorage({ host: 'myhosturl' });
```
So that we can allow for more options to be customized down the road. WDYT?